### PR TITLE
Fixed discount label to return null if not available

### LINKED ISSUE #4018: Word 'Discount' is duplicated + discount amount is displayed without symbol minus
**Description**:
Word 'Discount' is duplicated + discount amount is displayed without symbol minus ('-')

**Expected behavior**:
'Discount: -{discountAmount}' is displayed

**To Reproduce**:
1. Log in
2. Open My orders
3. Open order where cart price rule (without coupon) is applied to order

**Actual result**:
Word 'Discount' is duplicated + discount amount is displayed without symbol minus ('-')
Screenshot 2022-01-04 at 08 14 50

### LINKED ISSUE #4166: Mobile: Order page look ugly if user open it in RTL mode and applied coupon with long name without spaces
**Description**:
Mobile: Order page look ugly if user open it in RTL mode and applied coupon with long name without spaces
Fix after #3751 and #4074

**Expected behavior**:
Order page has same layout if user open it in RTL mode and applied coupon with long name without spaces

**Screenshots, Video, Logs**
image

**To Reproduce**:
1. Open Sign in form
2. Log in
3. Open My orders
4. Open order

**Actual result**:
Order page look ugly if user open it in RTL mode and applied coupon with long name without spaces

### PR DESCRIPTION
**Pull request by SalimMersally**
**Problem:**
- Discount label return "Discount" if there is no label, expected to return null

**In this PR:**
- Fixed discount label to return null

**Related PR:**
- https://github.com/scandipwa/scandipwa/pull/4074
- Fixes https://github.com/scandipwa/scandipwa/issues/4018
- Fixes https://github.com/scandipwa/scandipwa/issues/4166

### CODE REVIEW

File: Model/Resolver/OrderTotal.php [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        return $allAppliedTaxOnOrders;
+    }
+
+    /**
+     * Return taxes applied to the current order
+     *
+     * @param OrderInterface $order
+     * @return array
+     */
+    private function getAppliedTaxesDetails(OrderInterface $order): array
```
Comment (AleksandrsKondratjevs) [inline Model/Resolver/OrderTotal.php]: Make private functions to public.

File: Model/Resolver/OrderTotal.php [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    {
+        $allAppliedTaxOnOrders = $this->getAllAppliedTaxesOnOrders($order);
+        $taxes = [];
+        foreach ($allAppliedTaxOnOrders as $appliedTaxes) {
```
Comment (AleksandrsKondratjevs) [inline Model/Resolver/OrderTotal.php]: add spaces after and before foreach, if, while and etc blocks if there is other operations.

File: composer.json [hunk 3]
```diff
@@ -16,5 +16,6 @@
         "psr-4": {
             "ScandiPWA\\SalesGraphQl\\": ""
         }
-    }
+    },
+    "version": "1.0.10"
```
Comment (AleksandrsKondratjevs) [inline composer.json]: Remove hardcoded version

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

File: Model/Resolver/OrderTotal.php [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+namespace ScandiPWA\SalesGraphQl\Model\Resolver;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\SalesGraphQl\Model\Resolver\OrderTotal as SourceOrderTotal;
+
```
Comment (aleksandrsm) [inline Model/Resolver/OrderTotal.php:22]: 1 newline

File: Model/Resolver/OrderTotal.php [hunk 5]
```diff
@@ -0,0 +1,209 @@
+<?php
```
Comment (aleksandrsm) [inline Model/Resolver/OrderTotal.php:1]: SW PWA docs - for the format, check other modelues

File: Model/Resolver/OrderTotal.php [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\SalesGraphQl\Model\Resolver\OrderTotal as SourceOrderTotal;
+
+
+/**
+ * Resolve order totals taxes and discounts for order
```
Comment (aleksandrsm) [inline Model/Resolver/OrderTotal.php:24]: Standard docs (with package class etc)
Also, please add a comment on which methods got changed in the override

File: Model/Resolver/OrderTotal.php (hunk 2 above)
Comment (aleksandrsm) [reply Model/Resolver/OrderTotal.php]: newlines before foreach/if/return & so on statements are also appreciated

File: Model/Resolver/OrderTotal.php [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+     *
+     * @param OrderInterface $order
+     * @return array
+     */
+    private function getDiscountDetails(OrderInterface $order): array
+    {
+        $orderDiscounts = [];
+        if (!($order->getDiscountDescription() === null && $order->getDiscountAmount() == 0)) {
+            $orderDiscounts[] = [
+                'label' => $order->getDiscountDescription(),
```
Comment (aleksandrsm) [inline Model/Resolver/OrderTotal.php:135]: as I understand, this is the main part that got changed? If so, please add a note to the method docs

File: Model/Resolver/OrderTotal.php [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+     * @param OrderInterface $order
+     * @return array
+     */
+    private function getShippingDiscountDetails(OrderInterface $order): array
+    {
+        $shippingDiscounts = [];
+        if (!($order->getDiscountDescription() === null && $order->getShippingDiscountAmount() == 0)) {
+            $shippingDiscounts[] =
+                [
+                    'label' => $order->getDiscountDescription() ?? __('Discount'),
```
Comment (aleksandrsm) [inline Model/Resolver/OrderTotal.php]: wouldn't the same problem exist here?

**Review by aleksandrsm — CHANGES_REQUESTED** [review]
see comments

**Comment by carinadues** [comment]
Commenting to follow up

**Review by aleksandrsm — APPROVED** [review]

**Pull request merged**